ssl/quic/quic_tserver.c

```diff
@@ -121,6 +125,9 @@ QUIC_TSERVER *ossl_quic_tserver_new(const QUIC_TSERVER_ARGS *args,
 
 err:
     if (srv != NULL) {
+        if (args->ctx != NULL)
+            SSL_CTX_free(srv->ctx);
+        SSL_free(srv->tls);
```

Comment:
Need to fixup ossl_quic_tserver_free as well in this case. But it may be better to just make TSERVER require a SSL_CTX and remove the code where it can allocate it internally.

Reply:
`ossl_quic_tserver_free` already looks correct to me. Although I have tweaked the above code (the condition on when to free the SSL_CTX was wrong). The idea here is that if an explicit SSL_CTX is provided then we use that, otherwise we create one. In either case the reference is owned by the tserver instance - unless something goes wrong during construction of the tserver - in which case ownership of the SSL_CTX remains with the caller.

Reply:
Ah, so it consumes the reference. OK.